rust/src/passenger.rs

```diff
@@ -0,0 +1,105 @@
+use std::sync::Arc;
+
+use pyo3::{
+    prelude::*,
+    types::{PyDict, PyType},
+};
+
+use crate::membership::Membership;
+use crate::type_aliases::*;
+
+#[pyclass]
+#[derive(Clone)]
+pub struct Passenger {
```

Comment:
@nreinicke here's one of the big things we were going to revisit on a re-write: [resources instead of passengers](https://github.com/NREL/hive/issues/12). is there anything we should be thinking about as we work on this rust implementation? like, for instance, maybe these live on SimulationState.{passengers|resources} instead of in their vehicle states?

Reply:
yeah good thinking on looking forward to using generic resources; we talked about this at the sprint and think we can set up the concept of the resource in rust and still keep the passenger in python for backwards compatibility.

thinking about where to keep these, it could make sense to store them in the simulation state and then just have vehicles keep track of them via a resource id (like how we do with bases that have stations).

Reply:
#### about the PR

sounds good. and, since this implementation is pretty much a record class with a constructor and some accessors, nothing really stands in the way of us doing that renaming later and dealing with the resource-specifics later too. to which i'm saying, this is probably just fine to call "done".

#### more fun thoughts

another thing that might live elsewhere, such as the Environment, is a table that provides "types" of resources, in much of the same way we currently store different Schedules, Mechatronics, etc. a ResourceType might contain behaviors related to the type of resource, such as Passengers or IceCreamCones or whatever. but we can totally add that stuff later.

Reply:
#### more exploring robs brain

then a ResourceType is a trait, subtypes implementing `fn returnable(): bool`, `fn refridgerated(): bool` or some other kinds of ways to mash up that information. or, do we use something like the FleetId relationships graph to describe these instead? like, instead of worrying about all kinds of fields and compatibility tests endogenously, we just do a quick thing like a membership check, but just slightly different logic (go from an `any` test to an `all` test) for some `RequirementsId` graph? 🤔

Reply:
yeah this is a good question and I'm not confident about which way might be the best way to go.. maybe it would help to enumerate all the different behaviors that a resource might have..

Reply:
sure. future work. so, let's accept this one for now then right? be my guest @nreinicke